tiledb/sm/query/writer.h

```diff
@@ -422,6 +489,25 @@ class Writer {
   Status prepare_full_tiles(
       const std::string& attribute, std::vector<Tile>* tiles) const;
 
+  /**
+   * Applicable only to write in global order. It prepares only full
+   * tiles, storing the last potentially non-full tile in
+   * `global_write_state->last_tiles_` as part of the state to be used in
+   * the next write invocation. The last tiles are written to storage
+   * upon `finalize`. Upon each invocation, the function first
+   * populates the partially full last tile from the previous
+   * invocation.
+   *
+   * @param attribute The attribute to prepare the tiles for.
+   * @param coord_dups The positions of the duplicate coordinates.
+   * @param tiles The **full** tiles to be created.
+   * @return Status
+   */
+  Status prepare_full_tiles(
```

Comment:
Why copy/paste these functions instead of a small check for `coord_dups.empty()` within a single function?

Reply:
Because we are adding an `if` check inside all `for/while` loops, which I expect to slow down the performance. If this is not true, I can avoid the code duplication.

Reply:
It would be a very predictable branch, but it would also be better to just duplicate the loops, one with a branch one without, than the whole functions IMO.

Reply:
Good point, I can address that.